### PR TITLE
Add `of_collection` method and upgrade pds.api-client dependency for API v1.5.0

### PR DESCRIPTION
**Pull request by tloubrieu-jpl**
## 🗒️ Summary
These changes aim at supporting the Seismic use case proposed by Francesco Civilini.

## ⚙️ Test Data and/or Report
To test run:

    pytest

## ♻️ Related Issues

fixes: https://github.com/NASA-PDS/updart/issues/22

### CODE REVIEW

File: src/pds/updart/client.py [hunk 1]
```diff
@@ -94,7 +135,7 @@ def __next__(self):
         :return: next filtered product
         """
         try:
-            return next(self._crt_page_iterator)
+            return self._get_new_product()
         except StopIteration:
-            # TODO: implement next page retrieval and return first element of next page
-            raise
+            self._init_new_page()
+            return self._get_new_product()
```
Comment (nutjob4life) [inline src/pds/updart/client.py:141]: This whole `_init_new_page`, `return _get_new_product`, and `try…except StopIteration` could be simplified to fewer lines using a generator with `yield`

**Review by nutjob4life — APPROVED** [review]
Consider using a generator with a `yield` statement in next version, otherwise approved!

File: src/pds/updart/client.py (hunk 1 above)
Comment (tloubrieu-jpl) [reply src/pds/updart/client.py:141]: Thanks @nutjob4life , I don't see immediately how that would work. We can go over that during a breakout session, I will schedule it for this Tuesday.

Comment (nutjob4life) [reply src/pds/updart/client.py:141]: Oh no! I should've kept my mouth shut! 😂

**Pull request merged**